connectors/es/sink.py

```diff
@@ -326,9 +331,14 @@ async def _deferred_index(self, lazy_download, doc_id, doc, operation):
             }
         )
 
-    async def run(self, generator):
+    async def run(self, generator, job_type):
         try:
-            await self.get_docs(generator)
+            if job_type == JobType.FULL:
```

Comment:
Should we use `match/case` here like we did in the `connectors` module for updating and accessing `last_sync_time` for consistency? Ofc nothing blocking.